apis/common/v1/connection_details.go

```diff
@@ -18,6 +18,12 @@ package v1
 
 import (
 	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// LabelKeyOwnerUID is the UID of the owner resource of a connection secret.
+	LabelKeyOwnerUID = "secret.crossplane.io/owner-uid"
```

Comment:
This seems like it would duplicate Kubernetes Secret controller references in some cases. I suppose we need to go this path if we want to be consistent though, given that some Kubernetes Secrets might live in different clusters, and obviously controller references don't exist in Vault. Perhaps a comment noting that would help future readers understand why we use labels for this?